workflow/scripts/modify_district_heat_share.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+import logging
+
+logger = logging.getLogger(__name__)
+import geopandas as gpd
+import pandas as pd
+from shapely.geometry import Point
+
+
+def cluster_egon(heat_techs):
+    """
+    Map NUTS3 regions of egon data to corresponding clusters according to
+    maximum overlap.
+    """
+
+    regions_onshore = gpd.read_file(snakemake.input.regions_onshore)
```

Comment:
normally better to move `snakemake`inputs outside the function, so that the function can be called also from other scripts, you can add the parameter `fn`to the function instead with

```
def cluster_egon(heat_techs, fn):
    regions_onshore = gpd.read_file(fn)
```